### PR TITLE
Add support for ECE Payment buttons on Pay for Order page

### LINKED ISSUE #3411: [ECE] Implement ECE for payment processing on pay for order page
### Description
Display the ECE button and process payment with the button on the 'pay for order' page for orders having `pending` payment status. 

### Acceptance Criteria
ECE buttons are rendered on the 'pay for order' page.
You can successfully process payment with the ECE button from the 'pay for order' page.
Both success and failure cases are handled including necessary notices.

### PR DESCRIPTION
**Pull request by james-allan**
Closes #3411

## Changes proposed in this Pull Request:

This PR adds support for processing Express payment button payments via the Pay for Order page. 

<p align="center">
Screenshot 2024-09-19 at 10 13 42 am
</br>
<sup>Google Pay and Apple Pay buttons on the <strong>Pay for Order</strong></sup>
</p>
 
## Testing instructions

**SETUP**

1. Enable **Apple Pay / Google Pay** in your Stripe plugin settings.
2. Enable the ECE feature flag.
   - Update the `_wcstripe_feature_ece` option. See `WC_Stripe_Feature_Flags::is_stripe_ece_enabled()`

**PAYMENTS**

1. Go to **WooCommerce → Orders** on your WP Admin dashboard.
4. Click the **Add order** button.
5. Add your current user as the customer. 
6. Add any items. 
7. Calculate totals.
8. Create the order.
9. In the order details section click the **"Customer payment page"** link. ([Screenshot](https://github.com/user-attachments/assets/e5e345aa-c17e-49ef-b93f-566f8b380b6b))
10. On the following page note that the Apple Pay and/or Google Pay buttons are shown.
11. Choose either Google Pay or Apple Pay
12. Confirm the modal that pops up has correct totals etc.
 
| Order totals | Apple Pay screens |
|--------|--------|
| Screenshot 2024-09-19 at 10 33 02 am | Screenshot 2024-09-19 at 10 28 53 amScreenshot 2024-09-19 at 10 28 47 am | 

13. Complete the payment. 
14. Confirm you are redirected to the order received page. 
15. In your Stripe Dashboard confirm the payment is completed and the total matches the order total.

<p align="center">
Screenshot 2024-09-19 at 10 40 18 am
</p>

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -130,7 +130,8 @@ If you get stuck, you can ask for help in the Plugin Forum.
 
 = 8.8.0 - xxxx-xx-xx =
 * Add - Implemented the "Update all subscriptions payment methods" checkbox on My Account → Payment methods for UPE payment methods.
-* Add - Add support for the new Stripe Checkout Element on the shortcode checkout page. 
+* Add - Add support for the new Stripe Checkout Element on the shortcode checkout page.
```
Comment (james-allan) [inline readme.txt:135]: This change is just a blank space at the end of the line which is being removed.

File: client/entrypoints/express-checkout/styles.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#wc-stripe-express-checkout-element iframe {
+	max-width: unset;
```
Comment (james-allan) [inline client/entrypoints/express-checkout/styles.scss:2]: This fixes an issue on all shortcode powered pages and follows the same approach WooPayments took in https://github.com/Automattic/woocommerce-payments/pull/9094 and discussed with Stripe internallly here: p1720833500937719-slack-C9976E5MJ

**PAY FOR ORDER**
<p align="center">
Screenshot 2024-09-26 at 4 44 10 pm
</br>
<sup>Gap on the right hand side of the payment elements.</sup>
</p>

**CHECKOUT**
<p align="center">
Screenshot 2024-09-26 at 5 20 45 pm
</p> 


**CART**
<p align="center">
Screenshot 2024-09-26 at 5 11 45 pm
</p>

File: readme.txt (hunk 1 above)
Comment (mattallan) [reply readme.txt:135]: Weird that GH isn't showing the whitespace being removed!

**Review by mattallan — APPROVED** [review]
Really well put together PR @james-allan ! 💯 

I've performed code review and the changes look good.
I've also tested the following flows/cases:

- [x] Paying for an order with virtual/downloadable products
- [x] Paying for an order with various shipping & tax totals
- [x] Paying for an order with/without prices includes tax
- [x] Paying for an order with discounts applied
- [x] Paying for an order with multiple products
- [x] Paying for an order with mix of variable/simple products
- [x] Confirm orders have correct payment method title
- [x] Order under minimum amount (0.20). Confirmed buttons don’t show.
- [x] Test Apple Pay and Google Pay

---

Something I noticed is the Pay for Order doesn't add the quantity label like we do for the product page.

I.e. 
| Pay for Order | Product Page |
|--------|--------|
| image | image| 

This is a very minor detail and I noticed WooPayments does the same thing so happy to run with it as is.

**Comment by james-allan** [comment]
> Something I noticed is the Pay for Order doesn't add the quantity label like we do for the product page.

oh neat. I'll file a separate issue for this because I think that's a good enhancement.

**Pull request merged**